### PR TITLE
disttask: enlarge max-concurrent-task and optimize scheduler/executor loop

### PR DESCRIPTION
**Pull request by D3Hunter**
### What problem does this PR solve?

Issue Number: ref #49008

Problem Summary:

### What changed and how does it work?
- change WaitTask/refreshTask(both scheduler and executor)/GetTaskExecInfoByExecID to query task base only to reduce overhead
- change MaxConcurrentTask = 16
### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)

before this pr, if we run `BenchmarkSchedulerOverhead` with task-meta size = 4m and max-concurrent-task 4/8/16/32, cpu/mem is:
image
after(**The initial/middle/tail spike is caused by create-task and switch-2-next-step.**):
image

- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: pkg/util/misc.go [hunk 1]
```diff
@@ -110,15 +110,16 @@ func Recover(metricsLabel, funcInfo string, recoverFn func(), quit bool) {
...
-	if recoverFn != nil {
-		recoverFn()
-	}
 	logutil.BgLogger().Error("panic in the recoverable goroutine",
 		zap.String("label", metricsLabel),
 		zap.String("funcInfo", funcInfo),
 		zap.Any("r", r),
 		zap.Stack("stack"))
 	metrics.PanicCounter.WithLabelValues(metricsLabel).Inc()
+
```
Comment (D3Hunter) [inline pkg/util/misc.go:119]: to have log printed in case `recoverFn` blocks

File: pkg/disttask/framework/scheduler/scheduler.go [hunk 2]
```diff
@@ -134,11 +134,15 @@ func (s *BaseScheduler) GetTask() *proto.Task {
 // refreshTask fetch task state from tidb_global_task table.
 func (s *BaseScheduler) refreshTask() error {
 	task := s.GetTask()
-	newTask, err := s.taskMgr.GetTaskByID(s.ctx, task.ID)
+	// we only refresh the base fields of task to reduce memory usage, other fields
+	// must be maintained in memory by the scheduler itself.
```
Comment (D3Hunter) [inline pkg/disttask/framework/scheduler/scheduler.go:138]:
```suggestion
	// must be maintained in memory by the scheduler itself.
	// TODO it's possible we have a stale task meta on network partition, get task meta when needed??
```

File: pkg/disttask/framework/scheduler/scheduler.go [hunk 3]
```diff
@@ -134,11 +134,16 @@ func (s *BaseScheduler) GetTask() *proto.Task {
...
+	// TODO it's possible we have a stale task meta on network partition, get task meta when needed??
```
Comment (D3Hunter) [inline pkg/disttask/framework/scheduler/scheduler.go:139]: will fix it later

**Review by okJiang — APPROVED** [review]

**Review by ywqzzy — APPROVED** [review]

**Pull request merged**